google_rest_api_generator/src/lib.rs

```diff
@@ -164,23 +164,21 @@ impl APIDesc {
             .map(|method| method_actions::generate(method, &self.params));
         info!("outputting");
         quote! {
-            fn auth_token() -> &'static str {
-                #auth_token
-            }
-
             pub mod schemas {
                 #(#schemas_to_create)*
             }
             pub mod params {
                 #(#params_to_create)*
             }
-            pub struct Client{
+            pub struct Client<A> {
                 reqwest: ::reqwest::Client,
+                auth: ::std::sync::Mutex<A>,
             }
-            impl Client {
-                pub fn new() -> Self {
-                    Client{
+            impl<A> Client<A> {
```

Comment:
I know it's best practice to limit trait bounds to the minimal scope that they're necessary, but in this case I would like to restrict the impl to A: yup_oauth::GetToken. This is mostly to make sure the documentation for Client is intuitive. People will see new accepts A and I want it to be immediately obvious what A needs to be

Reply:
That's actually an oversight. Fixed now